src/Adafruit_si7021.cpp

```diff
@@ -150,18 +178,24 @@ void Adafruit_Si7021::readSerialNumber(void) {
   if (!gotData)
     return; // error timeout
 
-  sernum_a = Wire.read();
-  Wire.read();
-  sernum_a <<= 8;
-  sernum_a |= Wire.read();
-  Wire.read();
-  sernum_a <<= 8;
-  sernum_a |= Wire.read();
-  Wire.read();
-  sernum_a <<= 8;
-  sernum_a |= Wire.read();
-  Wire.read();
-
+  //receive data into buffer to simplify CRC calculation
+  uint8_t buf[8];
+  for(auto a=0;a<8; a++){
+      buf[a] = Wire.read();
+  }
+  sernum_a = 0;
+  for(auto a=0; a<4;a++){
+      sernum_a <<= 8;
+      sernum_a |= buf[a*2]; //skip over checksum bytes
+  }
+  // test crc
+  for(auto a= 0; a<4; a++){
+      if(calcCrc((uint8_t*)&buf[a*2],1) != buf[(a*2)+1]){
+          log_e("CRC mismatch on byte %d of First Serial number read. actual=0x%02x calc=0x%02x",(3-a),buf[(a*2)+1],
```

Comment:
dont use log_e - have a debug #ifdef if you need one and serial.prints

Reply:
```c++
#ifndef ARDUINO_ARCH_ESP32
 #define log_e()
#endif 
```

I see most of your code return quietly on errors do you actually want `Serial.prints()` with the ESP32 debug can be turned off at compile time.  Do you want equivalent messages that will work with Arduino AVR?